### ISSUE TITLE
Feature: User Sign-Up with Email Verification

### ISSUE THREAD

**Issue by Al-ArafatSiddique** [Al-ArafatSiddique/issue-template#1]
#### **Objective**
Implement a user sign-up feature that collects necessary user information, validates the password, and ensures email verification through an OTP system before account creation is finalized.

---

#### **Fields to Collect**
- **Name** (string): User's full name.
- **Email** (string): User's email address (must be unique).
- **Password** (string): Must meet the following validation standards:
  - Minimum 8 characters
  - At least 1 uppercase letter
  - At least 1 lowercase letter
  - At least 1 numeric character
  - At least 1 special character (e.g., !@#$%^&*)
- **Confirm Password** (string): Must match the password.

---

#### **Tasks Checklist**
- [ ] Create a sign-up form with the following fields: name, email, password, confirmPassword.
- [ ] Validate the password based on the criteria mentioned above.
- [ ] Ensure the confirm password field matches the password.
- [ ] Send a One-Time Password (OTP) to the user’s email upon sign-up submission.
- [ ] Implement an email verification process:
  - OTP should be valid for 24 hours.
  - Provide the user with an option to **resend OTP** if the email is missed.
- [ ] Finalize the account creation only after successful OTP verification.

---

#### **Scope**
- **In Scope**:
  - Create the sign-up feature.
  - Email validation and OTP-based verification.
  - Password strength validation.
  - Resend OTP functionality.
- **Out of Scope**:
  - Forgot password or login functionality.

---

#### **Success Criteria**
The feature will be considered successful if:
- The user is able to sign up with a valid email, password, and confirm password.
- The password meets the specified validation criteria.
- The OTP is sent to the user's email, valid for 24 hours, and can be resent if necessary.
- The account is created only after the OTP verification.

---

#### **Challenges/Assumptions**
- Ensure emails are delivered reliably to prevent verification issues.
- Handle cases where users don’t receive the email and request to resend OTP.
- OTP should expire after 24 hours, and a new OTP should be generated on resending.

---

#### **Implementation Steps**
1. Create a form with input fields for name, email, password, and confirm password.
2. Validate the form inputs, ensuring:
   - The email is unique and correctly formatted.
   - Password meets strength criteria and matches confirm password.
3. Send an OTP to the email provided.
4. Create a verification system for users to input the OTP.
   - OTP should expire after 24 hours.
   - Provide a "Resend OTP" button for missed emails.
5. Complete the sign-up process after the OTP is verified.

---

#### **Expected Outcome**
- Successful sign-up process with email verification.
- Passwords are validated and securely stored.
- Users can resend OTP if necessary and complete the sign-up process after verification.

---

#### **Timeline**
Provide an estimated timeline for the feature’s implementation (e.g., 1 week).

---

#### **Additional Context**
- Use libraries like Nodemailer for email and OTP generation.
- Ensure secure password hashing and storage using best practices like bcrypt or Argon2.
- Use react-hook-form to construct the sign up form
- Use zod for validation rules and infer types from that

---
